Transport `Razor.Utilities.Shared` and its dependencies

jjonescz opened the pull request:
Related: https://github.com/dotnet/sdk/pull/31508, https://github.com/dotnet/sdk/pull/31510

DustinCampbell commented on src/Compiler/tools/Microsoft.AspNetCore.Mvc.Razor.Extensions.Tooling.Internal/Microsoft.AspNetCore.Mvc.Razor.Extensions.Tooling.Internal.csproj:
```diff
@@ -13,6 +13,7 @@
     <ProjectReferemce Include="..\..\Microsoft.AspNetCore.Mvc.Razor.Extensions\src\Microsoft.AspNetCore.Mvc.Razor.Extensions.csproj" ReferenceOutputAssembly="false" />
 
     <Content Include="$(ArtifactsDir)bin\Microsoft.AspNetCore.Mvc.Razor.Extensions\$(Configuration)\netstandard2.0\Microsoft.AspNetCore.Mvc.Razor.Extensions.dll" PackagePath="lib\$(TargetFramework)" />
+    <Content Include="$(ArtifactsDir)bin\Microsoft.AspNetCore.Mvc.Razor.Extensions\$(Configuration)\netstandard2.0\Microsoft.AspNetCore.Razor.Utilities.Shared.dll" PackagePath="lib\$(TargetFramework)" />
```
Are any of the dependencies of `Utilities.Shared` needed, like `Microsoft.Extensions.ObjectPool`?

chsienki replied:
Yeah I would think we'd need MS.E.ObjectPool too.

jjonescz replied:
Makes sense, thanks!

jjonescz commented:
Verified locally that this should fix the errors in SDK:
1. Run `Microsoft.NET.Sdk.Razor.Tests` - seeing the errors.
2. Packaged Razor: `.\eng\common\build.ps1 -restore -build -pack`
3. Added `<add key="local" value="C:\razor\artifacts\packages\Debug\NonShipping" />` to SDK's `NuGet.config`.
4. Changed the Razor transport package versions to `7.0.0-dev` in SDK's `eng\Versions.props`.
5. Rebuild the SDK and re-run the tests - no errors.

chsienki commented on src/Compiler/tools/Microsoft.AspNetCore.Mvc.Razor.Extensions.Tooling.Internal/Microsoft.AspNetCore.Mvc.Razor.Extensions.Tooling.Internal.csproj:
```diff
@@ -10,7 +10,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <ProjectReferemce Include="..\..\Microsoft.AspNetCore.Mvc.Razor.Extensions\src\Microsoft.AspNetCore.Mvc.Razor.Extensions.csproj" ReferenceOutputAssembly="false" />
```
Lol, I fixed a bunch of these when we migrated over to a single repo. They 'worked' by virtue of those projects being built already when this gets built, and the project not actually using the contents other than via content. At this point I'm not sure if it was just a bad copy-paste or something deliberate? Either way 😆
